rotary/rules/annotation.smk

```diff
@@ -0,0 +1,399 @@
+          --data_dir {params.db} --override > {log} 2>&1
+        rm -r {output.outdir}/tmp
+        """
+
+
+rule run_gtdbtk:
+    input:
+        genome="{sample}/annotation/dfast/{sample}_genome.fna",
+        setup_finished=os.path.join(DB_DIR_PATH,"checkpoints","GTDB_" + VERSION_GTDB_COMPLETE + "_validate"),
+        ref_msh_file=os.path.join(DB_DIR_PATH,"GTDB_" + VERSION_GTDB_COMPLETE + '_mash','gtdb_ref_sketch.msh')
+    output:
+        batchfile=temp("{sample}/annotation/gtdbtk/batchfile.tsv"),
+        annotation="{sample}/annotation/gtdbtk/{sample}_gtdbtk.summary.tsv",
+        outdir=directory("{sample}/annotation/gtdbtk/run_files")
+    conda:
+        "../envs/gtdbtk.yaml"
+    log:
+        "{sample}/logs/annotation/gtdbtk.log"
+    benchmark:
+        "{sample}/benchmarks/annotation/gtdbtk.txt"
+    params:
+        db_dir=directory(os.path.join(DB_DIR_PATH,"GTDB_" + VERSION_GTDB_COMPLETE)),
+        genome_id='{sample}',# Get sample name from wildcards.
+        gtdbtk_mode="--full_tree" if config.get("gtdbtk_mode") == "full_tree" else ""
+    threads:
+        config.get("threads",1)
+    shell:
+        """
+        printf "{input.genome}\t{params.genome_id}\n" > {output.batchfile}
+        GTDBTK_DATA_PATH={params.db_dir} \
+        gtdbtk classify_wf --batchfile {output.batchfile} --out_dir {output.outdir} {params.gtdbtk_mode} \
+          --mash_db {input.ref_msh_file} --cpus {threads} --pplacer_cpus {threads} > {log} 2>&1
+        head -n 1 {output.outdir}/gtdbtk.*.summary.tsv | sort -u > {output.annotation}
+        tail -n +2 {output.outdir}/gtdbtk.*.summary.tsv >> {output.annotation}
+        """
+
+rule run_checkm2:
+    input:
+        genome="{sample}/annotation/dfast/{sample}_genome.fna",
+        install_finished=os.path.join(DB_DIR_PATH,"checkpoints","checkm2")
+    output:
+        quality_report="{sample}/annotation/checkm/{sample}_checkm_quality_report.tsv",
+        outdir=directory("{sample}/annotation/checkm/")
+    log:
+        "{sample}/logs/annotation/checkm.log"
+    benchmark:
+        "{sample}/benchmarks/annotation/checkm.txt"
+    conda:
+        "../envs/checkm2.yaml"
+    params:
+        db=directory(os.path.join(DB_DIR_PATH,"checkm2"))
+    threads:
+        config.get("threads",1)
+    shell:
+        """
+        mkdir -p {output.outdir}
+        checkm2 predict --threads {threads} --input {input.genome} --output-directory {output.outdir} > {log} 2>&1
+        mv {output.outdir}/quality_report.tsv {output.quality_report}
+        """
+
+
+if POLISH_WITH_SHORT_READS == True:
+    # TODO - clarify name compared to previous mapping step
+    rule calculate_final_short_read_coverage:
+        input:
+            qc_short_r1="{sample}/qc/{sample}_qc_R1.fastq.gz",
+            qc_short_r2="{sample}/qc/{sample}_qc_R2.fastq.gz",
+            dfast_genome="{sample}/annotation/dfast/{sample}_genome.fna"
+        output:
+            mapping=temp("{sample}/annotation/coverage/{sample}_short_read.bam"),
```

Comment:
@jmtsuji Sounds like a good idea. Will do.

Reply:
Moved to https://github.com/rotary-genomics/rotary/issues/118

Reply:
Sounds good to continue in #119! (I fixed what I think was a typo in your last post... I think you were discussing 119 rather than 118) Resolving this conversation for now.